internal/aws/common/aws.go

```diff
@@ -247,3 +247,97 @@ func VpcBySubnet(subnetId string) (string, error) {
 	vpcId := subnets.Subnets[0].VpcId
 	return *vpcId, nil
 }
+
+func getSubnetsRouteTable(vpcId string, subnetsRouteTable map[string]*ec2.RouteTable, nextToken **string) (err error) {
+	svc := connectors.GetAWSSession().EC2
+
+	routeTablesOutput, err := svc.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("vpc-id"),
+				Values: []*string{
+					&vpcId,
+				},
+			},
+		},
+		NextToken: *nextToken,
+	})
+
+	if err != nil {
+		return
+	}
+
+	*nextToken = routeTablesOutput.NextToken
+
+	for _, routeTable := range routeTablesOutput.RouteTables {
+		for _, association := range routeTable.Associations {
+			if association.SubnetId != nil {
+				subnetsRouteTable[*association.SubnetId] = routeTable
+				break
+			}
+		}
+	}
+	return
+}
+
+func getSubnetWithIdenticalRouteTable(vpcId, subnetId string, subnetsRouteTable map[string]*ec2.RouteTable, nextToken **string) (additionalVpcSubnet string, err error) {
+	svc := connectors.GetAWSSession().EC2
+
+	subnetsOutput, err := svc.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("vpc-id"),
+				Values: []*string{
+					&vpcId,
+				},
+			},
+		},
+		NextToken: *nextToken,
+	})
+
+	if err != nil {
+		return
+	}
+
+	*nextToken = subnetsOutput.NextToken
+
+	for _, subnet := range subnetsOutput.Subnets {
+		if *subnet.SubnetId == subnetId {
+			continue
+		}
+		if subnetsRouteTable[*subnet.SubnetId] == subnetsRouteTable[subnetId] {
+			additionalVpcSubnet = *subnet.SubnetId
+			return
+		}
+	}
+	return
+}
+
+func GetAdditionalVpcSubnet(vpcId, subnetId string) (additionalVpcSubnet string, err error) {
+	subnetsRouteTable := make(map[string]*ec2.RouteTable)
+	var nextToken *string
+
+	err = getSubnetsRouteTable(vpcId, subnetsRouteTable, &nextToken)
+	if err != nil {
+		return
+	}
+	for nextToken != nil {
+		err = getSubnetsRouteTable(vpcId, subnetsRouteTable, &nextToken)
```

Comment:
this should be a loop inside the function(or recursive), you don't know that this is just one continuation. 
and single function that does all the job, instead of handling pagination inside consumer function